pkg/brokerapi/broker.go

```diff
@@ -332,34 +334,21 @@ func (b Broker) Update(rctx *reqcontext.ReqContext, instanceID, serviceID, oldPl
 		return res, err
 	}
 
-	slaChangePermitted := func() bool {
-		instanceSLA := instance.Labels.SLA
-		newPlanSLA := np.Labels.SLA
-		instancePlanSize := instance.Labels.PlanSize
-		newPlanSize := np.Labels.PlanSize
-		instanceService := instance.Labels.ServiceID
-		newPlanService := np.Labels.ServiceID
-
-		// switch from redis to mariadb not permitted
-		if instanceService != newPlanService {
-			return false
-		}
-		// xsmall -> large not permitted, only xsmall <-> xsmall-premium
-		if instancePlanSize != newPlanSize {
-			return false
-		}
-		if instanceSLA == crossplane.SLAPremium && newPlanSLA == crossplane.SLAStandard {
-			return true
-		}
-		if instanceSLA == crossplane.SLAStandard && newPlanSLA == crossplane.SLAPremium {
-			return true
-		}
-		return false
+	cmp, err := p.Cmp(*np)
+	if err != nil {
+		return res, ErrPlanChangeNotPermitted
```

Comment:
Maybe add a clarifying comment here to note which conditions this return covers.

```suggestion
		// The new plan has a different service, unknown service, or unknown size.
		// This change is not permitted.
		return res, ErrPlanChangeNotPermitted
```